prime-dotnet-webapi/Controllers/AdminsController.cs

```diff
@@ -98,5 +100,21 @@ public async Task<ActionResult<Admin>> GetAdminById(int adminId)
 
             return Ok(ApiResponse.Result(admin));
         }
+
+        // GET: api/Admins/embedded-metabase-url
+        /// <summary>
+        /// Returns a time sensitive metabase embedded url string
+        /// </summary>
+        [HttpGet("embedded-metabase-url", Name = nameof(GetMetabaseEmbeddedString))]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<Admin>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<string>> GetMetabaseEmbeddedString()
+        {
+            var token = await Task.Run(() => _metabaseService.BuildMetabaseEmbeddedString());
+
+            return Ok(ApiResponse.Result(token));
```

Comment:
All the processing here is entirely synchronous, so we can make this controller method synchronus as well. Its not required to make controller methods async.